zerver/lib/queue.py

```diff
@@ -52,8 +52,14 @@ def _get_parameters(self) -> pika.ConnectionParameters:
         # it doesn't make sense with BlockingConnection
         credentials = pika.PlainCredentials(settings.RABBITMQ_USERNAME,
                                             settings.RABBITMQ_PASSWORD)
+        tcp_options = None
+        if self.rabbitmq_heartbeat == 0:
+            tcp_options = dict(TCP_KEEPIDLE=60,
+                               TCP_KEEPINTVL=10,
+                               TCP_KEEPCNT=9)
```

Comment:
We should provide a comment explaining the purpose of the change here.  It sounds like we're replacing 7200s / 75s with 60s / 10s?  That seems really aggressive.
```
tabbott@coset:~$ for i in /proc/sys/net/ipv4/tcp_keepalive_*; do echo $i $(cat $i); done
/proc/sys/net/ipv4/tcp_keepalive_intvl 75
/proc/sys/net/ipv4/tcp_keepalive_probes 9
/proc/sys/net/ipv4/tcp_keepalive_time 7200
```

Sending a keepalive packet every 10s seems quite aggressive; given the original issue report of connections being dropped after O(15 minutes), I think we should test doing something like `TCP_KEEPIDLE=300` with no other changes -- i.e. just changing the start of TCP keepalives to 5 minutes, rather than the default of 2 hours.  

And then we can write a nice comment explaining this:
```
In cases where we've disabled RabbitMQ's heartbeat, the only keepalive on this connection is the 
TCP keepalive (defaults: `/proc/sys/net/ipv4/tcp_keepalive_*`).  On most Linux systems, the default
is to start sending keepalive packets after TCP_KEEPIDLE (7200 seconds) of inactivity; after that point,
it send them every TCP_KEEPINTVL (typically 75s).  Some Kubernetes / Docker Swarm networks 
can kill "idle" TCP connections after as little as ~15 minutes of inactivity.  To avoid this killing our RabbitMQ connections, we set TCP_KEEPIDLE to something significantly below 15 minutes.
```